src/pydap/handlers/dap.py

```diff
@@ -258,15 +262,24 @@ def __iter__(self):
         # download and unpack data
         r = requests.get(self.url, stream=True)
         r.raise_for_status()
-        stream = StreamReader(r.iter_content(BLOCKSIZE))
 
-        # strip dds response
-        marker = b'\nData:\n'
-        buf = []
-        while ''.join(buf) != marker:
-            chunk = stream.read(1)
-            buf.append(chunk)
-            buf = buf[-len(marker):]
+        # Fast forward past the DDS header
+        # the pattern could span chunk boundaries though so make sure to check
+        i = r.iter_content(1024)
+        previous_chunk = b''
+        this_chunk = b''
+        pattern = b'Data:\n'
+        while not re.search(pattern, previous_chunk + this_chunk):
+            previous_chunk = this_chunk
+            this_chunk = next(i)
+        end_chunk = (previous_chunk + this_chunk).split(b'Data:\n', 1)[1]
```

Comment:
If you save the result of `re.search`, you can get the index on which to split directly with `m.end()`.